xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamPVRManager.cpp

```diff
@@ -268,15 +268,20 @@ bool CDVDInputStreamPVRManager::PrevChannel(bool preview/* = false*/)
 bool CDVDInputStreamPVRManager::SelectChannelByNumber(unsigned int iChannelNumber)
 {
   PVR_CLIENT client;
+  CPVRChannelPtr currentChannel(g_PVRManager.GetCurrentChannel());
+  CFileItemPtr item(g_PVRChannelGroups->Get(currentChannel->IsRadio())->GetSelectedGroup()->GetByChannelNumber(iChannelNumber));
```

Comment:
BTW we are in `SelectChannelByNumber` and you can only select a channel by it's channel number if the channel is part of the current select group because the channel numbers are not unique across all groups.